### PR TITLE
[FIX] web_decimal_numpad: extend FieldMonetary

### LINKED ISSUE #862: [11]web_decimal_numpad_dot and expenses field dont work like intended
in expenses on unit field the numpad dot remain a dot instead of comma

image

### PR DESCRIPTION
**Pull request by chienandalu**
- FieldMonetary wasn't extended thus it didn't have decimal numpad feature.
- Fix #862

cc @Tecnativa

### CODE REVIEW

File: web_decimal_numpad_dot/static/src/js/numpad_dot.js [hunk 1]
```diff
@@ -19,6 +19,40 @@ odoo.define("web_decimal_numpad_dot.FieldFloat", function (require) {
                 ? ":" : translation._t.database.parameters.decimal_point;
         },
 
+        numpad_dot_replace: function (event) {
```
Comment (yajo) [inline web_decimal_numpad_dot/static/src/js/numpad_dot.js]: To keep it DRY, please create a new object such as this:

```js
var NumpadDotReplaceMixin = {
   ...
}

basic_fields.FieldFloat.include(NumpadDotReplaceMixin);
basic_fields.FieldMonetary.include(NumpadDotReplaceMixin);

return {
    NumpadDotReplaceMixin: NumpadDotReplaceMixin,
}
```

**Review by yajo — CHANGES_REQUESTED** [review]
Interesting, it seems in Odoo v11 they stopped making `FieldMonetary` a subclass from `FieldFloat`, and that's why it wasn't working. Now both are completely independent.

Nice, just change it a bit as explained below. Thanks!

**Comment by chienandalu** [comment]
@Yajo Thanks for the tip! Changes done!

File: web_decimal_numpad_dot/static/src/js/numpad_dot.js [hunk 2]
```diff
@@ -39,5 +39,12 @@ odoo.define("web_decimal_numpad_dot.FieldFloat", function (require) {
             to = from + point.length
             this.$input.prop("selectionStart", to).prop("selectionEnd", to);
         },
-    });
+    }
```
Comment (yajo) [inline web_decimal_numpad_dot/static/src/js/numpad_dot.js]: Missing `;`

**Review by yajo — APPROVED** [review]

**Review by astirpe — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**